testing/00_title.md

```diff
@@ -44,162 +73,235 @@ plugins: RevealMarkdown, RevealChalkboard, RevealHighlight, RevealMath.KaTeX, Re
+  </div>
+  <img src="./files/metrics.jpg" width="300" style="margin-left: 60px">
+</div>
 
+  
 
 ---
 
 <!-- .slide: data-state="standard" data-background="./files/whitebg.png" -->
 
-# PyTest
-
-- recommended python testing framework
-- [docs.pytest.org](https://docs.pytest.org/en/7.3.x/)
-
-![](.files/pytest_logo.svg)
+## Misleading metrics
+
+Consider this function...
+```python
+def fahrenheit_to_celsius(temp_list, converted_temps=[]):
+    for temp in temp_list:
+        temp_c = (temp - 32.0) * (5.0/9.0)
+        converted_temps.append(temp_c)
+    return converted_temps
+```
+
+...and these tests with 100% coverage:
+<!-- .element: class="fragment" data-fragment-index="2" -->
+```python
+assert fahrenheit_to_celsius([32.0, 77.0]) == [0.0, 25.0]
+assert (fahrenheit_to_celsius([100], converted_temps = [0.0, 25.0])
+        == [0.0, 25.0, 37.77777777777778])
+```
+<!-- .element: class="fragment" data-fragment-index="2" -->
+
+Q: Can you think of a problem that will not be caught?
+<!-- .element: class="fragment" data-fragment-index="3" -->
 
 ---
 
 <!-- .slide: data-state="standard" data-background="./files/whitebg.png" -->
 
-## Write Code
 
+## Writing a test file
+
+Create a file called test_example.py ...
 <pre><code class="bash" style="overflow: hidden;" data-trim class="bash" data-line-numbers>
 $ mkdir pytest-example
 $ cd pytest-example
+$ nano test_example.py
 </code></pre>
 
 <div class="fragment">
-Creating a file <code>example.py</code> containing
+... and add the function and tests from before:
 <pre><code class="python" style="overflow: hidden;" data-trim class="bash" data-line-numbers>
-def add(a, b):
-    return a + b
+def fahrenheit_to_celsius(temp_list, converted_temps=[]):
+    for temp in temp_list:
+        temp_c = (temp - 32.0) * (5.0/9.0)
+        converted_temps.append(temp_c)
+    return converted_temps
 &nbsp;
 &nbsp;
-def test_add():  # Special name!
-    assert add(2, 3) == 5  # What's `assert`? 🤔
-    assert add('space', 'ship') == 'spaceship'
+def test_convert():  # Special name!
+    assert fahrenheit_to_celsius([32.0, 77.0]) == [0.0, 25.0]
+    assert (fahrenheit_to_celsius([100], converted_temps = [0.0, 25.0])
+            == [0.0, 25.0, 37.77777777777778])
 </code></pre>
 </div>
 
-<div class="fragment">
-Chat with the python shell about <code>assert</code> ...
-</div>
-<div class="fragment">
-<pre><code class="python" style="overflow: hidden;" data-trim class="bash" data-line-numbers>
->>> assert 1==1  # passes
->>> assert 1==2  # throws error
-Traceback (most recent call last):
-  File "&lt;stdin&gt;", line 1, in &lt;module&gt;
-AssertionError
-</code></pre>
-
-</div>
-
 ---
 
 <!-- .slide: data-state="standard" data-background="./files/whitebg.png" -->
 
-## Test!
+## Using pytest to run tests in Python
 
-<pre><code style="overflow: hidden;" data-trim class="bash" data-line-numbers="1|1-9">
-$ pytest example.py
+[Pytest](https://docs.pytest.org/en/7.3.x/) is the recommended python testing framework
```

Comment:
Maybe mention other frameworks, e.g. Junit in java, etc. to make it language-agnostic.